public/decorators/agg_configs.js

```diff
@@ -0,0 +1,78 @@
+import _ from 'lodash'; // TODO: refactor lodash dependencies
+import * as prov from 'ui/vis/agg_configs';
+
+export function decorateAggConfigs(Private) {
+  // eslint-disable-next-line import/namespace
+  const AggConfigs = prov.AggConfigs || Private(prov.VisAggConfigsProvider);
+
+  /**
+   * Recursively removes undefined values from object.
+   * @param {*} obj
+   */
+  function removeEmptyValues(obj) {
+    return Object.keys(obj)
+      .filter(k => obj[k] != null) // Filters out undefined and null objects
+      .reduce((newObj, k) => {
+        // Recursive call for arrays
+        if (Array.isArray(obj[k])) return { ...newObj, [k]: obj[k].map(removeEmptyValues) };
```

Comment:
Does it work on a multidimensional array?

Reply:
No, but the expected input will always be the `dsl` object from Kibana. Example:

```
{
  "3": {
    "terms": {
      "field": "machine.os.raw",
      "size": 5,
      "order": {
        "_count": "desc"
      },
      "missing": "not available"
    },
    "aggs": {
      "2": undefined,
      "4": {
        "terms": {
          "field": "extension.raw",
          "size": 5,
          "order": {
            "_count": "desc"
          },
          "missing": "not available"
        }
      }
    }
  }
}
```

Reply:
:+1: